tests/Unit/PrestaShopBundle/Routing/Converter/RouterProviderTest.php

```diff
@@ -407,4 +442,68 @@ private function buildRouteCollection(array $routes)
 
         return $routeCollection;
     }
+
+    /**
+     * @dataProvider getRoutes
+     */
+    public function testRouteWithoutFeatureFlagShouldNeverCallDatabase(
+        Route $route,
+        bool $expected
+    ): void {
+        $router = new RouterProvider(
+            $this->createMock(RouterInterface::class),
+            $featureFlagRepository = $this->createMock(FeatureFlagRepository::class)
+        );
+
+        // assert no calls to repository
+        $featureFlagRepository
+            ->expects(self::never())
+            ->method('isEnabled')
+        ;
+
+        $router->isLegacyRoute($route);
```

Comment:
I don't think this method should be public, but you can still validate this internal behaviour If you build this test based on the `getLegacyRoutes` method you can do the same, it's just that you assert the returned list instead of a specific one directly